src/scheduler/job_info.c

```diff
@@ -5161,6 +5146,19 @@ static int cull_running_jobs(resource_resv *job, void *arg)
 				}
 			}
 			break;
+		case INSUFFICIENT_SERVER_RESOURCE:
+		case INSUFFICIENT_QUEUE_RESOURCE:
+			policy = inp->job->server->policy;
+			/* Check for resources in the resources line that are not RASSN resources.
+			 * RASSN resources are accumulated across the select.
+			 * This means all jobs will have them, and it invalidates this check.
+			 */
+			if (resdef_exists_in_array(policy->resdef_to_check, inp->err->rdef) &&
```

Comment:
Shouldn't this be resdef_to_check_rassn?  We want to look at flag=q resources that are not flag=qh.

Reply:
Well, actually I think it wouldn't matter because the error is specific to server/queue resource. And we are checking that the resource is not one of the rassn resources that get translated to select-specification. The check was there to make sure that the resource also exists in resdef_to_check list.
Using resdef_to_check_rassn might speed things a little because in most cases it will be a subset of resdef_to_check itself. So it will be a smaller array to search into.
OR, I can remove this check for resdef_to_check completely because if is_ok_to_run is giving a resource definition in error structure then it better be in resdef_to_check.
What do you think?

Reply:
I think you are right, remove the check for resdef_to_check.  You are right that if is_ok_to_run() is returning a resource, it will be in resdef_to_check.